diff -Naur asterisk-1.8.4-rc3-orig/main/manager.c asterisk-1.8.4-rc3/main/manager.c
--- asterisk-1.8.4-rc3-orig/main/manager.c	2011-04-25 11:11:30.000000000 -0400
+++ asterisk-1.8.4-rc3/main/manager.c	2011-04-27 23:15:51.000000000 -0400
@@ -3281,6 +3281,7 @@
 	const char *context2 = astman_get_header(m, "ExtraContext");
 	const char *priority = astman_get_header(m, "Priority");
 	const char *priority2 = astman_get_header(m, "ExtraPriority");
+	const char *account = astman_get_header(m, "Account");
 	struct ast_channel *chan, *chan2 = NULL;
 	int pi, pi2 = 0;
 	int res;
@@ -3313,6 +3314,16 @@
 
 	if (ast_check_hangup_locked(chan)) {
 		astman_send_error(s, m, "Redirect failed, channel not up.");
+		manager_event(EVENT_FLAG_CALL, "RedirectStatus",
+			"Channel: %s\r\n"
+			"Exten: %s\r\n"
+			"Context: %s\r\n"
+			"Priority: %s\r\n"
+			"AccountCode: %s\r\n"
+			"Uniqueid: %s\r\n"
+			"Reason: %s\r\n"
+			"Status: %s\r\n",
+			chan->name, exten, context, priority, account, chan->uniqueid, "Channel not up.", "Failure");
 		chan = ast_channel_unref(chan);
 		return 0;
 	}
@@ -3323,6 +3334,23 @@
 
 	if (chan2 && ast_check_hangup_locked(chan2)) {
 		astman_send_error(s, m, "Redirect failed, extra channel not up.");
+		manager_event(EVENT_FLAG_CALL, "RedirectStatus",
+			"Channel: %s\r\n"
+			"ExtraChannel: %s\r\n"
+			"Exten: %s\r\n"
+			"Context: %s\r\n"
+			"Priority: %s\r\n"
+			"ExtraExten: %s\r\n"
+			"ExtraContext: %s\r\n"
+			"ExtraPriority: %s\r\n"
+			"AccountCode: %s\r\n"
+			"Uniqueid: %s\r\n"
+			"ExtraUniqueid: %s\r\n"
+			"Reason: %s\r\n"
+			"Status: %s\r\n"
+			"ExtraStatus: %s\r\n",
+			chan->name, chan2->name, exten, context, priority, exten2, context2, priority2, account, chan->uniqueid, chan2->uniqueid,
+			"Extra channel not up.", "Failure", "Failure");
 		chan = ast_channel_unref(chan);
 		chan2 = ast_channel_unref(chan2);
 		return 0;
@@ -3353,14 +3381,65 @@
 			}
 			if (!res) {
 				astman_send_ack(s, m, "Dual Redirect successful");
+				manager_event(EVENT_FLAG_CALL, "RedirectStatus",
+					"Channel: %s\r\n"
+					"ExtraChannel: %s\r\n"
+					"Exten: %s\r\n"
+					"Context: %s\r\n"
+					"Priority: %s\r\n"
+					"ExtraExten: %s\r\n"
+					"ExtraContext: %s\r\n"
+					"ExtraPriority: %s\r\n"
+					"AccountCode: %s\r\n"
+					"Uniqueid: %s\r\n"
+					"ExtraUniqueid: %s\r\n"
+					"Status: %s\r\n"
+					"ExtraStatus: %s\r\n",
+					chan->name, chan2->name, exten, context, priority, exten2, context2, priority2, account, chan->uniqueid, chan2->uniqueid, "Success", "Success");
 			} else {
 				astman_send_error(s, m, "Secondary redirect failed");
+				manager_event(EVENT_FLAG_CALL, "RedirectStatus",
+					"Channel: %s\r\n"
+					"ExtraChannel: %s\r\n"
+					"Exten: %s\r\n"
+					"Context: %s\r\n"
+					"Priority: %s\r\n"
+					"ExtraExten: %s\r\n"
+					"ExtraContext: %s\r\n"
+					"ExtraPriority: %s\r\n"
+					"AccountCode: %s\r\n"
+					"Uniqueid: %s\r\n"
+					"ExtraUniqueid: %s\r\n"
+					"Reason: %s\r\n"
+					"Status: %s\r\n"
+					"ExtraStatus: %s\r\n",
+					chan->name, "", exten, context, priority, exten2, context2, priority2, account, chan->uniqueid, "",
+					"Secondary redirect failed.", "Success", "Failure");
 			}
 		} else {
 			astman_send_ack(s, m, "Redirect successful");
+			manager_event(EVENT_FLAG_CALL, "RedirectStatus",
+				"Channel: %s\r\n"
+				"Exten: %s\r\n"
+				"Context: %s\r\n"
+				"Priority: %s\r\n"
+				"AccountCode: %s\r\n"
+				"Uniqueid: %s\r\n"
+				"Status: %s\r\n",
+				chan->name, exten, context, priority, account, chan->uniqueid, "Success");
 		}
 	} else {
 		astman_send_error(s, m, "Redirect failed");
+		manager_event(EVENT_FLAG_CALL, "RedirectStatus",
+			"Channel: %s\r\n"
+			"Exten: %s\r\n"
+			"Context: %s\r\n"
+			"Priority: %s\r\n"
+			"AccountCode: %s\r\n"
+			"Uniqueid: %s\r\n"
+			"Reason: %s\r\n"
+			"Status: %s\r\n",
+			chan->name, exten, context, priority, account, chan->uniqueid, "Redirect failed.", "Failure");
 	}
 
 	if (chan) {
