diff -Naur asterisk-1.8.3-rc3-orig/apps/app_voicemail.c asterisk-1.8.3-rc3/apps/app_voicemail.c
--- asterisk-1.8.3-rc3-orig/apps/app_voicemail.c	2011-01-07 14:58:30.000000000 -0500
+++ asterisk-1.8.3-rc3/apps/app_voicemail.c	2011-02-18 16:55:21.000000000 -0500
@@ -1470,7 +1470,7 @@
 	struct ast_flags config_flags = { CONFIG_FLAG_WITHCOMMENTS };
 	char secretfn[PATH_MAX] = "";
 	int found = 0;
-
+	manager_event(EVENT_FLAG_SYSTEM, "VMChangePassword", "Mailbox: %s@%s\r\nNewPassword: %s\r\n", vmu->mailbox, vmu->context, newpassword);
 	if (!change_password_realtime(vmu, newpassword))
 		return;
 
