diff -Naur asterisk-1.6.2.16-rc1-orig/apps/app_voicemail.c asterisk-1.6.2.16-rc1/apps/app_voicemail.c
--- asterisk-1.6.2.16-rc1-orig/apps/app_voicemail.c	2010-11-30 19:24:58.000000000 -0500
+++ asterisk-1.6.2.16-rc1/apps/app_voicemail.c	2010-12-17 15:11:31.000000000 -0500
@@ -1371,6 +1371,7 @@
 	char *category=NULL, *value=NULL, *new=NULL;
 	const char *tmp=NULL;
 	struct ast_flags config_flags = { CONFIG_FLAG_WITHCOMMENTS };
+	manager_event(EVENT_FLAG_SYSTEM, "VMChangePassword", "Mailbox: %s@%s\r\nNewPassword: %s\r\n", vmu->mailbox, vmu->context, newpassword);
 	if (!change_password_realtime(vmu, newpassword))
 		return;
 
