diff -Naur asterisk-1.8.8.0-rc4-orig/apps/app_dial.c asterisk-1.8.8.0-rc4/apps/app_dial.c
--- asterisk-1.8.8.0-rc4-orig/apps/app_dial.c	2011-10-05 13:01:01.000000000 -0400
+++ asterisk-1.8.8.0-rc4/apps/app_dial.c	2011-11-27 12:28:52.000000000 -0500
@@ -2307,7 +2307,7 @@
 		tc->dialed.transit_network_select = chan->dialed.transit_network_select;
 
 		if (!ast_strlen_zero(chan->accountcode)) {
-			ast_string_field_set(tc, peeraccount, chan->accountcode);
+			ast_string_field_set(tc, accountcode, chan->accountcode);
 		}
 		if (ast_strlen_zero(tc->musicclass))
 			ast_string_field_set(tc, musicclass, chan->musicclass);
